### ISSUE TITLE
Cannot wire the ladder

### ISSUE THREAD

**Issue by iridescent-lightning** [Regalis11/Barotrauma#13109]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Basically if you assign a ConnectionPanel Component to an item that is also a ladder, you can't open the connection panel either in game or sub editor. Therefore the ladders can't be wired.

Why do I need this feature: I made teleport doors that allow player to be teleported into different rooms. The doors are connected by wire so door A connects to door B. If door A is clicked, player teleported to door B. It seems like entering an inner room. 

However, bots can't use the door because it doesn't support waypoints. Two fake ladders connecting by wire can help solve this problem. Because a ladder with waypoint, will trick bots to use it even though the ladder doesn't lead to anywhere, therefore I can teleport the user.

<Item name="teleportladder" identifier="teleportladder" tags="ladder" resizevertical="false" scale="0.5" allowrotatingineditor="false" allowedlinks="teleportladder">
  <Sprite texture="Content/Items/Ladder/ladder.png" depth="0.51" sourcerect="34,8,22,240"/>
  <ConnectionPanel selectkey="Action" canbeselected="true" msg="ItemMsgRewireScrewdriver" hudpriority="10" canbepicked="false">
      <GuiFrame relativesize="0.2,0.32" minsize="400,350" maxsize="480,420" anchor="Center" style="ConnectionPanel" />
      <RequiredItem identifier="screwdriver" type="Equipped" />
      <output name="signal_out" displayname="connection.signalout" />
      <output name="signal_in" displayname="connection.signalin" />
    </ConnectionPanel>
  <Ladder canbeselected="true" msg="ItemMsgClimbSelect">
    <BackgroundSprite texture="Content/Items/Ladder/ladder.png" depth="0.05" sourcerect="8,8,26,240" origin="1.0,0.0"/>
  </Ladder>
  
  <trigger x="-40" y="20" width="90"/>
</Item>

### Reproduction steps

1. make a faker ladder.
2. go to the editor and put one
3. enter the wire mode and try to open the connection panel
4. nothing happens

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.1.19.3 (Treacherous Tides Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

This is currently does not work because there's some special logic in ladders that makes them behave a little differently from other items (the character basically selects them in a slightly different way than other items, to allow having both the ladders and another device you're interacting with active at the same time).

I'm not entirely sure I understand why you need the wiring though, what does the wire do? What do you mean by the door not supporting waypoints?

**Comment by iridescent-lightning** [comment]
![20231202221956_1](https://github.com/Regalis11/Barotrauma/assets/94741028/e17d0129-5542-47ba-8a88-495a32e5ba9f)
![20231202222031_1](https://github.com/Regalis11/Barotrauma/assets/94741028/351c38d7-241f-452c-97e7-f12ba077369a)
Thanks for the reply. As you can see in this new crewroom. The dorm is isolated from the main corridor. The teleport doors are connected by white wires. The teleport method is achieved via Lua. The signal sender will be teleported. As these doors are merely wire-connected items, they can't have waypoints on them so the bots neither can enter nor leave the areas. I did some experiments, if two separated ladders have waypoints pass through them, the bot will try to climb them. If ladders can be wired, I can instantly teleport the climber to the other side. If you put ladders behind the doors, it will look like the bots are using teleporting doors.

**Comment by BlackMoons** [comment]
If the wires are just to tell your LUA script what doors are linked, would linking would? (Space+mouseclick linking)

**Comment by iridescent-lightning** [comment]
> If the wires are just to tell your LUA script what doors are linked, would linking would? (Space+mouseclick linking)

The key problem is that bots won't interact with doors.

**Comment by BlackMoons** [comment]
Yes but can you link the ladders? Or use a button near a door? Iv noticed they do press buttons wired to(? In proximity of?) doors.

**Comment by Regalis11** [comment]
Thank you for the extra info!

> Yes but can you link the ladders? Or use a button near a door? Iv noticed they do press buttons wired to(? In proximity of?) doors.

I was wondering this too: if the wire is used to determine where the "other side" of the teleport is, could that be done by linking the items instead?

I suppose it could be possible for us to make the ladders wireable too, but I feel like that might turn out to require reworking the logic more than what I'd be comfortable with to support such a strange (although very creative!) use case. So I hope this could be solved by linking the items, or some other type of solution, instead.

**Comment by iridescent-lightning** [comment]
Yeah, thanks. After adjusting the XML, the doors can link to the ladders. I think I can teleport the character. Is there a way to tell if a specific ladder is being climbed? And also to get the climber?

**Comment by iridescent-lightning** [comment]
> > Is there a way to tell if a specific ladder is being climbed? And also to get the climber?
> 
> I'm not sure what parts of the code you can access via the Lua mod, but assuming Lua allows it, one way to do this would be to check whether the `SelectedSecondaryItem` of any character equals to the ladder.

I actually did it with C#. But it only works for controlled character. Once bots touch the ladder the game instantly crushed. Not reference to a object crush. Doesn't work if I tried to stun bots first.
